WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,198 @@
+            let size = delegate.collectionView?(collectionView, layout: self, sizeForItemAt: indexPath)
+        else {
+            return CGSize(width: itemSize.width, height: cellHeight)
+        }
+
+        return CGSize(width: size.width, height: cellHeight)
+    }
+
+    override open func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        if collectionView?.traitCollection.horizontalSizeClass == .regular {
+            return centeredLayoutAttributesForElements(in: rect)
+        }
+
+        var layoutAttributes = [UICollectionViewLayoutAttributes]()
+
+        for attribute in self.layoutAttributes {
+            if attribute.frame.intersects(rect) {
+                layoutAttributes.append(attribute)
+            }
+        }
+
+        return layoutAttributes
+    }
+
+    private func centeredLayoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var rows = [Row]()
+        var rowY: CGFloat = CGFloat.greatestFiniteMagnitude
+
+        // Create an array of "rows" based on the y positions
+        for attribute in layoutAttributes {
+            if attribute.frame.intersects(rect) == false {
+                continue
+            }
+
+            let minY = attribute.frame.minY
+
+            if rowY != minY {
+                rowY = minY
+
+                rows.append(Row(itemSpacing: itemSpacing, isRightToLeft: isRightToLeft))
+            }
+
+            rows.last?.add(attribute: attribute)
+        }
+
+        return rows.flatMap { (row: Row) -> [UICollectionViewLayoutAttributes] in
+            return row.centeredLayoutAttributesIn(width: maxContentWidth)
+        }
+    }
+
+    override open func layoutAttributesForItem(at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        return layoutAttributes[indexPath.row]
+    }
+}
+
+// MARK: - Row Centering Helper Class
+private class Row {
+    var layoutAttributes = [UICollectionViewLayoutAttributes]()
+    let itemSpacing: CGFloat
+    let isRightToLeft: Bool
+
+    init(itemSpacing: CGFloat, isRightToLeft: Bool) {
+        self.itemSpacing = itemSpacing
+        self.isRightToLeft = isRightToLeft
+    }
+
+    /// Add a new attribute to the row
+    /// - Parameter attribute: layout attribute to be added
+    public func add(attribute: UICollectionViewLayoutAttributes) {
+        layoutAttributes.append(attribute)
+    }
+
+    /// Calculates a new X position for each item in this row based on a new x offset
+    /// - Parameter width: The total width of the container view to center in
+    /// - Returns: The new centered layout attributes
+    public func centeredLayoutAttributesIn(width: CGFloat) -> [UICollectionViewLayoutAttributes] {
+        let centerX = (width - rowWidth) * 0.5
+
+        var offset = isRightToLeft ? width - centerX : centerX
+
+        for attribute in layoutAttributes {
```

Comment:
No need to change but just wanted to mention that this could be
`layoutAttributes.forEach { attribute in...`